ui/components/grant.js

```diff
@@ -30,6 +18,23 @@ export default function ({ id, modalOpen, handleCloseModal }) {
   const [role, setRole] = useState('view')
 
   const options = ['view', 'edit', 'admin', 'remove']
+  const emailList = [];
```

Comment:
In this case we should show the most "powerful" role for the user (from `view` to `admin`) vs simply taking the first. Subsequently we should fix this in the API (I've created https://github.com/infrahq/infra/issues/1804 to track)

A good approach could be to store a map `{email: role}` (example below) to display in the UI so users only see the "most powerful" role a user has. Removing their role in the UI should remove _all_ roles for this user even if we're only showing one in the UI.

Example:
```
{
  'jeff@infrahq.com': 'admin', 'view'
}
```